pulp_rpm/app/modulemd.py

```diff
@@ -2,14 +2,50 @@
 import os
 import tempfile
 
-from pulpcore.app.models.content import Artifact
+from pulpcore.app.models.content import Artifact, Content
 from pulp_rpm.app.constants import PULP_MODULE_ATTR, PULP_MODULEDEFAULTS_ATTR
 
 import gi
 gi.require_version('Modulemd', '2.0')
 from gi.repository import Modulemd as mmdlib  # noqa: E402
 
 
+def resolve_module_packages(version):
+    """
+    Decide which packages to add/remove based on modular data.
+
+    Args:
+        version (pulpcore.app.models.RepositoryVersion): current incomplete repository version
+    """
+    repo = version.repository
+    base_version = version.base_version
+    if base_version:
+        previous_version = base_version
+    else:
+        previous_version = repo.versions.get(number=repo.last_version - 1)
```

Comment:
in case this repo would be empty, we'd hit an error
```In [24]: x                                                                                                                                                                                                         
Out[24]: <ContainerRepository: ntkvb>

In [25]: x.last_version                                                                                                                                                                                            
Out[25]: 0

In [26]: x.last_version - 1                                                                                                                                                                                        
Out[26]: -1

In [27]: x.versions.get(number=-1)  

DoesNotExist: RepositoryVersion matching query does not exist.
```

Reply:
actually i think most of the tests fails because of this.

Reply:
@daviddavis , version-1 might not work if previous versions are removed.
I do it in this terrible way now:
```
    repo = version.repository
    base_version = version.base_version
    previous_version = base_version
    if not previous_version:
        repo_version_numbers = sorted(list(repo.versions.values_list("number", flat=True)))
        # if there are previous versions
        if len(repo_version_numbers) > 1:
            previous_version_number = repo_version_numbers[-2]
            previous_version = repo.versions.get(number=previous_version_number)
```